core/src/main/java/org/infinispan/transaction/xa/recovery/RecoveryManager.java

```diff
@@ -45,18 +50,22 @@
 
 
    /**
-    * Removes from the specified nodes (or all nodes if the value of 'where' is null) the recovery information associated with
-    * these Xids.
-    * @param where on which nodes should this be executed.
-    * @param xid the list of xids to be removed.
-    * @param sync execute sync or async (false)
-    * @param gtx
+    * Removes from the specified nodes (or all nodes if the value of 'where' is null) the recovery information
+    * associated with these Xids.
+    *
+    * @param where                   on which nodes should this be executed.
+    * @param xid                     the list of xids to be removed.
+    * @param sync                    execute sync or async (false)
+    * @param gtx                     the global transaction
+    * @param skipTxCompletionCommand {@code true} if it must skip the {@link org.infinispan.commands.remote.recovery.TxCompletionNotificationCommand}.
+    *                                Used when a partition happens.
     */
-   void removeRecoveryInformationFromCluster(Collection<Address> where, Xid xid, boolean sync, GlobalTransaction gtx);
+   void removeRecoveryInformationFromCluster(Collection<Address> where, Xid xid, boolean sync, GlobalTransaction gtx, boolean skipTxCompletionCommand);
```

Comment:
Ok, I  think I got confused with so many `removeRecoveryInformation[FromCluster]` methods :)

I think it may be a bit clearer if we renamed `removeRecoveryInformation` to `removeLocalRecoveryInformation` and stopped calling it from `removeRecoveryInformationFromCluster`, but it's ok either way.